homeassistant/components/smarla/switch.py

```diff
@@ -0,0 +1,94 @@
+"""Support for the Swing2Sleep Smarla switch entities."""
+
+from dataclasses import dataclass
+from typing import Any
+
+from pysmarlaapi import Federwiege
+from pysmarlaapi.federwiege.classes import Property
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import FederwiegeConfigEntry
+from .entity import SmarlaBaseEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmarlaSwitchEntityDescription(SwitchEntityDescription):
+    """Class describing Swing2Sleep Smarla switch entity."""
+
+    service: str
+    property: str
+
+
+SWITCHES: list[SmarlaSwitchEntityDescription] = [
+    SmarlaSwitchEntityDescription(
+        key="cradle",
+        name=None,
+        service="babywiege",
+        property="swing_active",
+    ),
+    SmarlaSwitchEntityDescription(
+        key="smartmode",
+        translation_key="smartmode",
+        service="babywiege",
+        property="smartmode",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: FederwiegeConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Smarla switches from config entry."""
+    federwiege: Federwiege = config_entry.runtime_data
+    async_add_entities(SmarlaSwitch(federwiege, desc) for desc in SWITCHES)
+
+
+class SmarlaSwitch(SmarlaBaseEntity, SwitchEntity):
+    """Representation of Smarla switch."""
+
+    entity_description: SmarlaSwitchEntityDescription
+    _property: Property
+    _attr_should_poll = False
```

Comment:
Does every entity for this integration get push updates? should we move the should poll to the base entity?